src/vis.ts

```diff
@@ -0,0 +1,38 @@
+import * as THREE from 'three';
+import * as plane from './plane';
+import {OrbitControls} from 'three/examples/jsm/controls/OrbitControls';
+
+export class Vis {
+  renderer: THREE.WebGLRenderer;
+  scene: THREE.Scene;
+  camera: THREE.PerspectiveCamera;
+  aircraft: THREE.Group;
+  camControls: OrbitControls;
+
+  constructor() {
+    const width = window.innerWidth,
+      height = window.innerHeight;
+
+    this.camera = new THREE.PerspectiveCamera(70, width / height, 0.01, 10);
+    this.camera.position.z = 1;
+
+    this.scene = new THREE.Scene();
+
+    this.aircraft = new THREE.Group();
+    this.aircraft.name = 'aircraft';
+    this.scene.add(this.aircraft);
+
+    this.renderer = new THREE.WebGLRenderer({antialias: true});
+    this.camControls = new OrbitControls(this.camera, this.renderer.domElement);
+    this.camControls.update();
+    this.renderer.setSize(width, height);
+    this.renderer.setAnimationLoop(this.animation);
+  }
+  animation = (time: DOMHighResTimeStamp) => {
```

Comment:
I believe that a lambda member of a class isn't any different from a class method in typescript. As such I'd recommend defining this as a class method instead:

`animation(time: DOMHighResTimeStamp) {`

Reply:
In order to reference the object's state I have to bind it to the function in some way. As I understand it, `this` isn't bound to the context a method is called from, so in calling a method form of animation from the constructor, it calls a generic animation with no context that contains `this`. In this instance, `this` is undefined. My understanding is still fuzzy, and this may be an XY problem, but there are a couple of ways I've found to solve this:

1. Manually bind the instance to the method when the callback is defined in the constructor, EG `this.renderer.setAnimationLoop(this.animation.bind(this));`
2. Use an arrow function. This (pun intended), for whatever reason, automatically binds to the parent scope.

I prefer method 2 as it isolates the scope binding "issue" (might just be a skill issue) to the implementation of the function that needs the scope bound, rather than relying on any future usage of the function to properly call `bind()` on the callback. I could be convinced otherwise.

Reply:
You're right about the binding, I assumed passing `this.animation` would work similarly in TS to how `this::animation` would work in Java.

There's an explanation I found useful for why the class field arrow function captures `this` [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions#cannot_be_used_as_methods).

In this situation, [Google style would recommend using an anonymous arrow function that calls a non-arrow function method on your class](https://google.github.io/styleguide/tsguide.html#arrow-functions-as-properties) in order to capture `this` when registering the callback while avoiding defining a method on your class that handles `this` differently from other methods.

The Google style guide does acknowledge the utility of class field arrow functions [when you need to be able to keep a reference to the callback so you can unregister it in the future](https://google.github.io/styleguide/tsguide.html#event-handlers).

Reply:
I currently don't intend to uninstall the event handler, so I think my usage does fall into the 4.5.9 case. I changed it regardless to build some muscle memory, and I think the arrow function as a property is a little nicer as it more explicitly shows the weirdly implicit argument setAnimationLoop passes. Thanks for the MDN link.